fashion-messaging-app-api/server.js

```diff
@@ -152,6 +167,58 @@ app.get('/photos', async (req, res) => {
   }
 });
 
+app.get('/recommendations', async (req, res) => {
+  try {
+    if (!req.session.user) {
+      return res.status(401).json({ error: 'Unauthorized' });
+    }
+
+    const lastSearch = await SearchResult.findOne({
+      where: { userId: req.session.user.id },
+      order: [['createdAt', 'DESC']],
+    });
+
+    const mostFrequentSearch = await SearchResult.findOne({
+      where: { userId: req.session.user.id },
+      order: [['searchCount', 'DESC']],
+    });
+
+    let lastSearchPhotos = [];
+    let mostFrequentSearchPhotos = [];
+
+    if (lastSearch) {
+      const response = await clientAPI.photos.search({ 
+        query: lastSearch.searchTerm, 
+        per_page: PHOTOS 
+      });
+      lastSearchPhotos = response.photos;
+
+    }
+
+    if (mostFrequentSearch) {
+      const response = await clientAPI.photos.search({ 
+        query: mostFrequentSearch.searchTerm, 
+        per_page: PHOTOS 
+      });
+      mostFrequentSearchPhotos = response.photos;
+    }
+
+let photos = [...lastSearchPhotos, ...mostFrequentSearchPhotos];
+photos = [...new Map(photos.map(photo => [photo.id, photo])).values()];
```

Comment:
This merges the lastSearched and most frequent searches into a photo array which I mapped through iand displayed in the recommendations page